client/state/sites/test/selectors.js

```diff
@@ -3354,23 +3354,7 @@ describe( 'selectors', () => {
 			expect( isMainNetwork ).to.equal( null );
 		} );
 
-		test( 'it should return `false` for multi-network sites', () => {
-			const state = createStateWithItems( {
-				[ siteId ]: {
-					ID: siteId,
-					URL: 'https://jetpacksite.me',
-					jetpack: true,
-					options: {
-						is_multi_network: true,
-					},
-				},
-			} );
-
-			const isMainNetwork = isJetpackSiteMainNetworkSite( state, siteId );
-			expect( isMainNetwork ).to.equal( false );
-		} );
-
-		test( 'it should return `true` for non multisite site', () => {
```

Comment:
It's interesting why it was designed in a way that it should return `true` for any non-multisite site. If site is not multi-site, it can't be main network site. Or are we missing some usage? Maybe it's just incorrect usage of naming.

Reply:
If you look at the two [related methods](https://github.com/Automattic/wp-calypso/blob/master/client/lib/site/jetpack.js#L56-L62) of `JetpackSite`, you can see that they work the same way as the Redux selectors after fixing.

I think that the incorrect behavoir was due to the [`canUpdateFiles`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/site/utils.js#L77) function in `client/lib/site/utils.js`. It uses the [`isMainNetworkSite`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/site/utils.js#L120) function in the same module.

Notice how that function is very different from the `JetpackSite.isMainNetworkSite` method, although it has the same name. What is really does is to "check a few network-related reasons why `canUpdateFiles` should return `false`". For example, when the site is a part of multi-network, or if it is a secondary site of some network. The name is very misleading.